packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.tsx

```diff
@@ -19,20 +25,54 @@ import DroppablePanel from '../droppable-panel';
 import SettingsContainer from '../settings-container';
 import messages from './messages';
 import { HIDDEN_COLUMNS_PANEL, SELECTED_COLUMNS_PANEL } from './constants';
+import { warning } from '@commercetools-uikit/utils';
+
+type TColumnData = {
+  key: string;
+  label: ReactNode;
+};
+
+type TColumnSettingsManagerProps = {
+  availableColumns: TColumnData[];
+  selectedColumns: TColumnData[];
+  onUpdateColumns: (updatedColums: TColumnData[]) => void;
+  areHiddenColumnsSearchable?: boolean;
+  searchHiddenColumns?: (searchTerm: string) => Promise<unknown>;
+  searchHiddenColumnsPlaceholder?: string;
+
+  onClose: (
+    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
+  ) => void;
+  primaryButton?: JSX.Element;
+  secondaryButton?: JSX.Element;
+  managerTheme?: 'light' | 'dark';
+};
+
+type TDroppableContainerProps = {
+  isDragging: boolean;
+};
+
+// TODO: Very hacky solution.
+// searchHiddenColums is only required if areHiddenColumnsSearchable is true
+// but that callback is used in AsyncSelectInput.loadOptions which is required
+// That property (loadOptions) is passed to react-select AsyncSelect where it's
+// declared as optional
+const noSearch = () => Promise.resolve();
```

Comment:
Not sure if this is the best solution...

`searchHiddenColums` is only required if `areHiddenColumnsSearchable` is true but that callback is used in `AsyncSelectInput.loadOptions` which is required.
That property (`loadOptions`) is passed to `react-select` `AsyncSelect` where it's declared as optional

Reply:
We can trigger a warning instead if areHiddenColumnsSearchable is true and searchHiddenColums is not a function.

Reply:
Btw, whenever, requiredIf is used in prop-types we use [warning](https://github.com/commercetools/ui-kit/blob/main/packages/utils/src/warning.ts) instead when migrating.

Reply:
@Rhotimee, @ddouglasz Do you mind if I look for some time in your calendars and schedule a meeting to discuss this one?

Reply:
sure, we can have a call tomorrow to discuss this.

Reply:
Sure

Reply:
As discussed, we agreed in using the `@ts-ignore` directive as it is safe for this use case and we avoid creating a dummy function.